### PR TITLE
Add staking precompile

### PR DESCRIPTION
**Pull request by codchen**
## Describe your changes and provide context
Add a precompile so that `delegate/redelegate/undelegate` can be made as EVM contract calls.

## Testing performed to validate your change
unit test && local sei with CLI

### CODE REVIEW

File: precompiles/staking/staking.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+func (p Precompile) RequiredGas(input []byte) uint64 {
+	methodID := input[:4]
+
+	if bytes.Equal(methodID, p.DelegateID) {
+		return 50000
+	} else if bytes.Equal(methodID, p.RedelegateID) {
+		return 70000
+	} else if bytes.Equal(methodID, p.UndelegateID) {
+		return 50000
+	}
```
Comment (philipsu522) [inline precompiles/staking/staking.go:92]: are these ever going to change?

File: precompiles/staking/staking_test.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+	"github.com/sei-protocol/sei-chain/x/evm/keeper"
+	"github.com/sei-protocol/sei-chain/x/evm/types"
+	evmtypes "github.com/sei-protocol/sei-chain/x/evm/types"
+	"github.com/sei-protocol/sei-chain/x/evm/types/ethtx"
+	minttypes "github.com/sei-protocol/sei-chain/x/mint/types"
+	"github.com/stretchr/testify/require"
+	tmtypes "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+func TestStaking(t *testing.T) {
```
Comment (philipsu522) [inline precompiles/staking/staking_test.go:31]: would suggest adding 1-2 tests with some more edge cases:
1. undelegating with no delegation
2. redelegating with no delegation

**Review by philipsu522 — APPROVED** [review]

File: precompiles/staking/staking.go (hunk 1 above)
Comment (codchen) [reply precompiles/staking/staking.go:92]: it may change if we change staking implementation (mainly want this to match cosmos staking gas cost), so I think keeping it as a constant should be fine?

File: precompiles/staking/staking_test.go (hunk 2 above)
Comment (codchen) [reply precompiles/staking/staking_test.go:31]: added
